web/src/stream_topic_history.ts

```diff
@@ -194,12 +214,16 @@ export class PerStreamHistory {
                 message_id,
                 pretty_name: topic_name,
                 historical: true,
+                // TODO(evy): what do we do for count here? it almost definitely isn't 0
+                // Should we try to calcualte it?
+                // Do we want to let it be optional, and then deal with that elsewhere?
+                count: 0,
```

Comment:
Open question here!

Reply:
[here](https://github.com/zulip/zulip/pull/28414/commits/487e58b8bb09f13339aad7520e258d574e59091c#r1440890083) Tim said

> I think the type here is intended to have the property that if `historical` is `true`, `count` is undefined -- we don't know. So one theory would be to define the `Topic` / `TopicHistoryEntry` type as a union of `historical: true` with no `count` field, and `historical: false` with a count field. #25438 will try to rework this, but I don't think we expect to without significant DB/API migrations break the invariant that server history entries don't come with a `count` for how many messages exist in the topic, so we should declare the types accordingly.

Reply:
switched to a union type